src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -43,8 +43,19 @@ public void TestEnvironmentProperty()
             environment.Remove("NewKey");
             Assert.Equal(CountItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // Exception thrown with invalid key
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
+            // The full .NET Framework uses CaseSensitiveStringDictionary, but we can't have an
+            // equivilent as we don't have access to the internal state of StringDictionary in .NET Core.
+            if (PlatformDetection.IsFullFramework)
```

Comment:
Without debugging, just looking at the StringDictionaryWithComparer code, which is what it uses on Windows, I'm not sure why it's not throwing when you insert the same key a second time. Is it because it's got the same vlaue? I don't see that...

I'm not sure what you mean by 'how to fix this' -- you mean fix core so it doesn't throw if hte value is the same?